File: Coding/recursion.py
# # def fact(n):
# #     if n<2:
# #         return 1
# #     return n*fact(n-1)
# # print(fact(5))


# # def fib(i):
# #     if i == 0 or i == 1:
# #         return i
# #     return fib(i - 1) + fib(i - 2)


# # print(fib(3))


# # isSorted
# def isSorted(arr):
#     if len(arr) < 2:
#         return 1
#     elif arr[0] > arr[1]:
#         return False
#     return isSorted(arr[1:])


# from array import array

# arr = array("i", [1, 2, 23, 4])
# print(isSorted(arr))


# # length of string
# def lenss(s):
#     if not len(s):
#         return 0
#     return 1 + lenss(s[1:])


# print(lenss("himanshu"))


def r(s):
    if s[0]=='x':
        for i in range(0,len(s)):
            s[i]=s[i+1];
    r(s[1:])
s="himanxxhushuxx"
r(s)
print(s)

File: Coding/recursion.cpp
// #include <bits/stdc++.h>
// using namespace std;
// int fact(int n)
// {
//     if (n < 2)
//     {
//         return n;
//     }
//     int smallAns = fact(n - 1);
//     return n * smallAns;
// }
// int fib(int n)
// {
//     if (n == 0 || n == 1)
//         return n;
//     return fib(n - 1) + fib(n - 2);
// }
// // for sum return n+smallAsn;
// bool isSorted(int arr[], int n)
// {
//     if (n == 1)
//         return 1;
//     if (arr[0] > arr[1])
//     {
//         return false;
//     }
//     return isSorted(arr + 1, n - 1);
// }
// int lens(string s)
// {
//     if (s[0] == '\0')
//     {
//         return 0;
//     }
//     return 1 + lens(s.substr(1));
// }
// int main()

// {
//     cout << lens("himanshu");
//     // cout << fact(3);
//     // cout << fib(3);
//     // int arr[] = {1, 2, 32, 4};
//     // int n = 4;
//     // cout << isSorted(arr, n);
//     return 0;
// }

#include <bits/stdc++.h>
using namespace std;
void ok(string &s)
{
    // string ss="";
    if (s.size() == 0)
    {
        return;
    }
    if (s[0] == 'x')
    {
        for (int i = 0; s[i] != '\0'; i++)
        {
            s[i] = s[i + 1];
        }
    }
    ok(s.substr(1));
    // ss=s;
}
int main()
{
    string s = "popuxhimanx";
    ok(s);
    cout << s;
    return 0;
}